File: 18-Command/src/go/light.go
package main

import "fmt"

// Light is the Receiver — it knows how to perform the real work.
type Light struct {
	location string
}

func NewLight(location string) *Light {
	return &Light{location: location}
}

func (l *Light) On() {
	fmt.Printf("  %s light is ON\n", l.location)
}

func (l *Light) Off() {
	fmt.Printf("  %s light is OFF\n", l.location)
}

File: 18-Command/src/rust/light.rs
/// Receiver — knows how to perform the real work.
pub struct Light {
    location: String,
}

impl Light {
    pub fn new(location: &str) -> Self {
        Light { location: location.to_string() }
    }

    pub fn on(&self) {
        println!("  {} light is ON", self.location);
    }

    pub fn off(&self) {
        println!("  {} light is OFF", self.location);
    }
}
